### PR TITLE
Fixes #12285: Change defaults for embedding xml and yaml

### PR DESCRIPTION
**Pull request by peckpeck**
https://www.rudder-project.org/redmine/issues/12285

### CODE REVIEW

File: rudder-agent/SOURCES/Makefile [hunk 1]
```diff
@@ -132,23 +132,29 @@ endif
...
 BUILD_DEPS := $(BUILD_DEPS) build-curl
 INSTALL_DEPS := $(INSTALL_DEPS) install-curl
 CURL_ARG := --with-libcurl=$(CURDIR)/build-curl/opt/rudder
+else
+CURL_ARG := --with-libcurl
 endif
-# default: embed libyaml
-ifneq (true,$(USE_SYSTEM_YAML))
+# default: do not embed libyaml
+ifeq (true,$(USE_SYSTEM_YAML))
```
Comment (amousset) [inline rudder-agent/SOURCES/Makefile]: why not ifneq ?

**Comment by peckpeck** [comment]
Commit modified

**Comment by peckpeck** [comment]
Commit modified

File: rudder-agent/SPECS/rudder-agent.spec [hunk 2]
```diff
@@ -51,28 +51,49 @@
...
-# no system perl on aix
+%else
+%define use_system_fusion true
+%endif
+
+# 1- AIX
 %if "%{?_os}" == "aix"
+# no system anything on aix
 %define use_system_perl false
+%define use_system_openssl false
```
Comment (amousset) [inline rudder-agent/SPECS/rudder-agent.spec]: missing new dependencies

**Comment by peckpeck** [comment]
Commit modified

**Pull request merged**